api/user/comments/model.py

```diff
@@ -1,21 +1,32 @@
 from dataclasses import asdict, dataclass
 
 from api.app import db
-from api.utils import generate_object_ID
+from api.utils import generate_object_ID, get_author_info
 
 
 @dataclass
 class Comment(db.Model):
     id: int = db.Column(db.String(50), primary_key=True, default="")
-    created: str = db.Column("created", db.String(20), nullable=False)
+    published: str = db.Column("published", db.String(20), nullable=False)
```

Comment:
shouldn't this be a datetime type?

```
# imports
from sqlalchemy.sql import functions
```

```suggestion
    published: str = db.Column("published", db.DateTime(), nullable=False, server_default=functions.now)
```
https://docs.sqlalchemy.org/en/20/core/functions.html#sqlalchemy.sql.functions.now

Reply:
ISO8601 (which includes timezone info) stamp is received in json. I had it as datetime earlier. String suffices for comparison operators.

Reply:
ISO8601 timestamps are directly string comparable. So I kept them as strings